### PR TITLE
🏑 Add `Status` Field Appointments Schema

### LINKED ISSUE #301: Add Status Field To Appointments Schema
- [x] Make the approved field a legacy field
- [x] Add a new field (Required) for the current status of the appointment
- [x] Update the form validation in `endpoint_utils` to support the new field for appointment status
- [x] Create a const object with 3 values { Pending, Rejected, Accepted } on backend/frontend
- [x] Update endpoints/frontend pages that deal with the state of appointment

### PR DESCRIPTION
**Pull request by Leundai**
## Description :star2:

Fixes #301 
- Added status field
- added consts to both frontend/backend
- updated endpoints
- updated api/frontend handling 

## TODOs :star:
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: frontend/src/utils/dateFormatting.js [hunk 1]
```diff
@@ -11,7 +12,9 @@ export const formatAppointments = (data) => {
     pending: [],
     past: [],
   };
-  const appointments = data.requests;
+  const appointments = data.requests.filter((elem) => {
+    return elem.status !== APPOINTMENT_STATUS.REJECTED;
+  });
```
Comment (kenetec) [inline frontend/src/utils/dateFormatting.js:17]: you can actually shorten this to `data.requests.filter(elem => elem.status !== APPOINTMENT_STATUS.REJECTED)`

**Review by kenetec — APPROVED** [review]
guccii

File: frontend/src/components/AppointmentInfo.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ function AppointmentInfo(props) {
 
   const getSubtext = (gender, organization) => {
     var subtextInfo = [gender];
-    if (organization != undefined) {
+    if (organization !== undefined) {
```
Comment (AngelaLuo49021) [inline frontend/src/components/AppointmentInfo.js:29]: good catch!

**Review by AngelaLuo49021 — APPROVED** [review]
